src/test/java/htsjdk/variant/variantcontext/writer/VCFWriterUnitTest.java

```diff
@@ -133,6 +137,49 @@ public void testBasicWriteAndRead(final String extension) throws IOException {
 
     }
 
+    /** test, using the writer and reader, that we can output and input a VCF body without problems */
+    @Test(dataProvider = "vcfExtensionsDataProvider")
+    public void testWriteAndReadVCFBody(final String extension) throws IOException {
+        final File fakeVCFFile = VariantBaseTest.createTempFile("testWriteAndReadVCFBody.", extension);
+        if (".vcf.gz".equals(extension)) {
+            new File(fakeVCFFile.getAbsolutePath() + ".tbi").deleteOnExit();
+        } else {
+            Tribble.indexFile(fakeVCFFile).deleteOnExit();
+        }
```

Comment:
It would be simpler to just create a temp directory (with deleteOnExit), and then just create the file in the temp dir. Then a lot of this code would be unnecessary.